shell/platform/fuchsia/flutter/tests/flatland_platform_view_unittest.cc

```diff
@@ -0,0 +1,1358 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <fuchsia/ui/composition/cpp/fidl.h>
+#include <fuchsia/ui/composition/cpp/fidl_test_base.h>
+#include <fuchsia/ui/input/cpp/fidl.h>
+#include <fuchsia/ui/input3/cpp/fidl.h>
+#include <fuchsia/ui/input3/cpp/fidl_test_base.h>
+#include <fuchsia/ui/scenic/cpp/fidl.h>
+#include <fuchsia/ui/views/cpp/fidl.h>
+#include <lib/async-loop/cpp/loop.h>
+#include <lib/async-loop/default.h>
+#include <lib/async/default.h>
+#include <lib/fidl/cpp/binding_set.h>
+#include <lib/ui/scenic/cpp/view_ref_pair.h>
+
+#include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "flutter/flow/embedded_views.h"
+#include "flutter/lib/ui/window/platform_message.h"
+#include "flutter/lib/ui/window/pointer_data.h"
+#include "flutter/lib/ui/window/viewport_metrics.h"
+#include "flutter/shell/common/context_options.h"
+#include "flutter/shell/platform/fuchsia/flutter/flatland_platform_view.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "fakes/focuser.h"
+#include "fakes/platform_message.h"
+#include "fakes/touch_source.h"
+#include "fakes/view_ref_focused.h"
+#include "flutter/shell/platform/fuchsia/flutter/surface.h"
+#include "flutter/shell/platform/fuchsia/flutter/task_runner_adapter.h"
+#include "platform/assert.h"
+#include "pointer_event_utility.h"
+
+namespace flutter_runner::testing {
+namespace {
+
+class MockExternalViewEmbedder : public flutter::ExternalViewEmbedder {
+ public:
+  SkCanvas* GetRootCanvas() override { return nullptr; }
+  std::vector<SkCanvas*> GetCurrentCanvases() override {
+    return std::vector<SkCanvas*>();
+  }
+
+  void CancelFrame() override {}
+  void BeginFrame(
+      SkISize frame_size,
+      GrDirectContext* context,
+      double device_pixel_ratio,
+      fml::RefPtr<fml::RasterThreadMerger> raster_thread_merger) override {}
+  void SubmitFrame(GrDirectContext* context,
+                   std::unique_ptr<flutter::SurfaceFrame> frame) override {
+    return;
+  }
+
+  void PrerollCompositeEmbeddedView(
+      int view_id,
+      std::unique_ptr<flutter::EmbeddedViewParams> params) override {}
+  SkCanvas* CompositeEmbeddedView(int view_id) override { return nullptr; }
+};
+
+class MockPlatformViewDelegate : public flutter::PlatformView::Delegate {
+ public:
+  void Reset() {
```

Comment:
I mean you're kinda asking me questions I don't know the answer to about code i didn't write, this class is unchanged from the GFX version of these tests. That being said, my read on this code is no, surface_ should not be reset because  its not transient state that would be changed multiple times in a single test. This is further evidenced by the assert in MockPlatformViewDelegate::OnPlatformViewCreated() which would insta-fail any test which attempted to reuse the MockPlatformViewDelegate with a new surface. I would prefer to take the "if it aint broke dont fix it" approach with this if you don't feel too strongly about it.

Reply:
Yeah, it's a tough position to be in! You've clearly thought about this more than I have, and everyone will appreciate the tests, so we're miles ahead already. Fine to submit as it is. 

But, if you have a minute, would be nice to  apply the campsite principle -  leaving a small comment to help the next confused author/reviewer. ;-)